File: increasing-triplet-subsequence/increasing-triplet-subsequence.go
func increasingTriplet(nums []int) bool {
    l := 10000000000
    r := 10000000000
    
    for _,num := range nums{
        if num <= l{
            l = num
        } else if num <= r{
            r = num
        } else{
            return true
        }
    }
    return false
}

File: increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums1 = nums2 = float('inf')
        
        for i in nums:
            if i <= nums1:
                nums1 = i
            elif i <= nums2:
                nums2 = i
            else:
                return True
        return False